feature-detects/mediastream.js

```diff
@@ -0,0 +1,13 @@
+define(['Modernizr'], function (Modernizr) {
+    Modernizr.addTest('mediarecorder', function(){
+        var isMediaRecorderSupported = false;
+
+        try {
```

Comment:
A number of browsers perform a lot slower when code has a try/catch block, so it is best to avoid it if possible. We are really checking if MediaRecorder is undefined or not, so a safer/more readable way to do this would be 

```js
typeof MediaRecorder !== undefined
```

or, since the `typeof` check would return either a string (which is truthy) if it exists,  or an actual `undefined` (which is falsey), a common shortening of this would be 


```js
typeof MediaRecorder
```